src/codegen/compile.ml

```diff
@@ -6602,6 +6848,19 @@ module MakeSerialization (Strm : Stream) = struct
           Tagged.allocation_barrier env ^^
           G.i Drop
         )
+      | Prim Region ->
+         read_alias env (Prim Region) (fun get_region_typ on_alloc ->
+          let (set_region, get_region) = new_local env "region" in
+          (* sanity check *)
+          get_region_typ ^^
+          compile_eq_const (Int32.neg (Option.get (to_idl_prim (Prim Region)))) ^^
+          E.else_trap_with env "deserialize_go (Region): unexpected idl_typ" ^^
+          Region.alloc env (compile_unboxed_const 0l) (compile_unboxed_const 0l) (Blob.lit env "") ^^ set_region ^^
+          on_alloc get_region ^^
+          get_region ^^ ReadBuf.read_word32 env get_data_buf ^^ Region.store_field env Region.id_field ^^
+          get_region ^^ ReadBuf.read_word32 env get_data_buf ^^ Region.store_field env Region.page_count_field ^^
```

Comment:
Potentially, it would be safer to offer one initialization function that takes all of these three field values.
This is because setting `vec_pages` is potentially critical if it would not be an initialzation (otherwise requiring a write barrier).